### ISSUE TITLE
Unable to decode the same object (Proto3)

### ISSUE THREAD

**Issue by charbgr** [square/wire#1648]
Hi there!

I may have fallen to a bug with Wire 3.2.2. Maybe it is related with https://github.com/square/wire/issues/1056 but not so sure! 
I am doing the following:

```kotlin
val subscription = Subscription(d = true)
val box = Box(subscription = subscription)
val boxEncoded = box.encode()
val boxDecoded = Box.ADAPTER.decode(boxEncoded)
```

and I got the following stacktrace: `Exception in thread "main" java.io.IOException: Expected to end at 20 but was 4`

The equivalent snippet with `Subscription` works great!

I have a repro project here: https://github.com/pavlospt/refactored-umbrella/tree/protobuf_issue (see latest commits)

The `Box`'s bytes are `2a122001`, and seems like `2001` are the `Subscription`’s bytes.

Am I doing something wrong?

**Comment by swankjesse** [comment]
proto3 is not yet supported. You wouldn't be using that by chance would you? Otherwise it's likely a bug where we're computing the size wrong.

**Comment by charbgr** [comment]
Based on this issue https://github.com/square/wire/issues/279, I thought I could use the wire `3.2.2` for proto3 (everything I need seems checked -dont care about Java). I think it has to do with computing the size wrong, thats why I opened the issue.

**Comment by oldergod** [comment]
No, proto3 isn't ready yet. We don't expect it to work yet.

**Comment by charbgr** [comment]
@oldergod This is resolved with `3.3.0` version! I am closing it!
